### PR TITLE
Doc: Add docs for Auditbeat add_session_metadata processor

### PR DESCRIPTION
**Pull request by karenzone**
Relates #40186

**PREVIEW:** https://beats_bk_40252.docs-preview.app.elstc.co/guide/en/beats/auditbeat/master/add-session-metadata.html

## Proposed commit message
Adds docs for configuring and enabling `add_session_metadata` processor for Auditbeat. This processor powers the Session View tool in the Elastic Security Platform. 

## ~ToDo:~ Find out how (if at all) the different doc location impacts our docs tooling. **DONE**
Most processors and docs live in `https://github.com/elastic/beats/tree/main/libbeat/processors`
The `add_session_metadata processor lives in `https://github.com/elastic/beats/tree/main/x-pack/auditbeat/processors/sessionmd`.

### Partner PR required to get docs building  (conf.yaml)
* https://github.com/elastic/docs/pull/3038

### Dependency to break free of gridlock created by changes in two repos: 
* https://github.com/elastic/beats/pull/40357

### CODE REVIEW

File: libbeat/docs/processors-list.asciidoc
Comment (karenzone) [inline libbeat/docs/processors-list.asciidoc:196]: Note to sel @karenzone: Don't forget these extra returns. Leaving in for now for easy spotting.

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (mjwolf) [inline x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:71]: There's also an option to set audit rules in a separate file. In the `auditbeat.yml` file the configuration would specify a file containing the audit rules, and that file should have the same rules as listed here.

For example,  `audit_rule_files: [ '${path.config}/audit.rules.d/*.conf' ]` in auditbeat.yml, and those files would contain the rules. This separate file is already an existing feature in auditbeat

I don't know if this is going to be too complicated and confusing here. Maybe above it could say something like
"Add audit rules in the modules configuration section of `auditbeat.yml` or audit rules file, depending on your configuration."

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (karenzone) [inline x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:36]:
```suggestion
```

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (karenzone) [inline x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:63]:
```suggestion
```

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (karenzone) [inline x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:63]: @mjwolf, how's this?

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (karenzone) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:67]: See change on lines 62-63.

**Comment by karenzone** [comment]
Please review and comment so that we can nail down the content.

**Comment by karenzone** [comment]
Docs-ci will fail with `ERROR: ../../libbeat/docs/processors-list.asciidoc: line 187: include file not found: /tmp/docsbuild/D3FctgCItb/beats/x-pack/auditbeat/processors/add_session_metadata/docs/add_session_metadata.asciidoc` until https://github.com/elastic/docs/pull/3038 has been merged

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (mjwolf) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:63]: > @mjwolf, how's this for looping in the config file option?

I think this is good, I would just change "\*the\* audit.rules.d config file" to "\*an\*  audit.rules.d config file", since there can be multiple files in that directory.

**Review by nick-alayil — COMMENTED** [review]
LGTM.  Thanks @mjwolf for your meticulous review and @karenzone for the excellent work on the documentation!

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (karenzone) [inline x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:62]:
```suggestion
. Add audit rules in the modules configuration section of `auditbeat.yml` or an
```

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (andrewkroh) [inline x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:31]: We should advise what is necessary in terms of permissions and OS config. Imaging you are a user trying to run this inside a container using least privileges. Do I need to be root? What capabilities are required? Do I need to have debugfs and tracefs mounted? etc.

**Review by benironside — APPROVED** [review]
LGTM! I like how it links to the session view docs in the intro.

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (karenzone) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:59]: @andrewkroh, is this question for @nick-alayil? @mjwolf? Either or both?

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (karenzone) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:31]: Is this question for @nick-alayil? @mjwolf? Either or both? :-)

Comment (nick-alayil) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:31]: The initial release of the auditbeat + session view was limited to VMs and bare metal environments. Support for Docker and Kubernetes is planned to follow. @karenzone we should mention this in the documentation.

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (nick-alayil) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:59]: > Is there use case for enriching FIM events that have a PID? 

In the context of enabling the session view, I don't think it's necessary. @mjwolf can confirm it. However, it made me wonder if we could display FIM alerts in the session view. That would be an interesting use case to discuss separately. Currently we do display file and network alerts from Defend datastream in session view.

**Comment by karenzone** [comment]
@karenzone Note to self:  Conflict resolution created by placeholder PR https://github.com/elastic/beats/pull/40357

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (mjwolf) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:59]: > > Is there use case for enriching FIM events that have a PID?
> 
> In the context of enabling the session view, I don't think it's necessary. @mjwolf can confirm it. However, it made me wonder if we could display FIM alerts in the session view. That would be an interesting use case to discuss separately. Currently we do display file and network alerts from Defend datastream in session view.

Currently FIM data won't power session view, but I agree maybe some of the extra enriched data could be useful to FIM. I think we should leave this as-is for now, and discuss changing it later

**Comment by karenzone** [comment]
run docs-build

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (karenzone) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:33]: We're careful not to make commitments for future development in format documentation.  I added a note that says "The current release of `add_session_metadata` processor for {auditbeat} limited to virtual machines (VMs) and bare metal environments."  This accurately describes current state while hinting that more enhancements are coming.

Comment (karenzone) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:33]: The multi-threaded comments are getting difficult to follow.   If I'm not mistaken, we don't have an answer to @andrewkroh's comment : 

>"We should advise what is necessary in terms of permissions and OS config. Imaging you are a user trying to run this inside a container using least privileges. Do I need to be root? What capabilities are required? Do I need to have debugfs and tracefs mounted? etc."

This is outside my area of expertise, and I'm not sure how to resolve this comment.  Can somebody help? 
Bonus points of you answer the question in the form of a github suggestion so that your intent for content is 100% clear.

**Comment by karenzone** [comment]
buildkite test this rebuild

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (mjwolf) [reply x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:33]:
```suggestion
==== Backends
* `auto` is the recommended setting, and will attempt to use `ebpf` first, falling back to `procfs` if necessary, ensuring compatibility even on systems without `ebpf` support.
* `ebpf` will collect process information with eBPF. This backend requires a system with Linux kernel 5.10.16 or above, kernel support for eBPF enabled, and auditbeat running as superuser.
* `procfs` will collect process information with the proc filesystem. This is compatible with older systems that may not support ebpf. To gather complete process info, auditbeat requires permissions to read all process data in procfs; for example, run as a superuser or have the `SYS_PTRACE` capability. 

==== Containers
If running auditbeat in a container, the container must run in the host's PID namespace. With the `auto` or `ebpf` backend, these host directories must also be mounted to the same path within the container: `/sys/kernel/debug`, `/sys/fs/bpf` 
```

This will probably need some editing, but this should be the information needed

**Comment by karenzone** [comment]
run docs-build rebuild

**Comment by karenzone** [comment]
#### Builds locally, but fails PR checks.

buildkite/docs-build-pr — Failed
```
INFO:build_docs:asciidoctor: ERROR: ../../libbeat/docs/processors-list.asciidoc: line 187: include file not found: /tmp/docsbuild/h360rSjkek/beats/x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
  | INFO:build_docs:asciidoctor: WARNING: invalid reference: add-session-metadata
```

**Notes to self @karenzone:**
* New x-pack docs location is present in `conf.yaml`:  https://github.com/elastic/docs/blob/master/conf.yaml#L1676-L1678
* Placeholder [PR](https://github.com/elastic/beats/pull/40357) passed CI and merged without error.  
   Nothing would be building yet because I haven't merged any includes. 
* This seems to point to error in `include` statement in this PR.   But then why would this work when I build locally???

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (dedemorton) [inline x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:10]: The conditions are set in the processor list, so you do not want to set them here.
```suggestion
```

File: x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc
Comment (dedemorton) [inline x-pack/auditbeat/processors/sessionmd/docs/add_session_metadata.asciidoc:89]:
```suggestion

```

**Comment by karenzone** [comment]
**Update:** After recent changes, this PR is passing docs-ci.  New `add_session_metadata` processor content is added to Auditbeat Ref, and is excluded for all other Beats references as intended.

**Comment by karenzone** [comment]
@mjwolf, thank you so much for answering remaining questions and suggesting some text.  I have committed your work with only minor edits. 

Please review and let's get this one published.

**Review by mjwolf — APPROVED** [review]
Looks good!

**Review by nick-alayil — APPROVED** [review]
LGTM. Thanks @karenzone and @mjwolf !!

**Review by andrewkroh — APPROVED** [review]

**Review by kilfoyle — APPROVED** [review]
LGTM! 🎸

**Comment by karenzone** [comment]
### Status

This PR touches every Beat, and requires approval from all Beats code owners.  We're missing approvals from obs-ds-hosted-services and @vihasmakwana. Please review so that we can get these docs published.

**Comment by pierrehilbert** [comment]
@karenzone Vihas is in PTOs until end of the week

**Review by kaiyan-sheng — APPROVED** [review]

**Comment by karenzone** [comment]
12686

**Comment by karenzone** [comment]
### Link: https://www.elastic.co/guide/en/beats/auditbeat/master/add-session-metadata.html

Screen Shot 2024-08-13 at 4 16 07 PM

**Pull request merged**
